apps/web/vibes/soul/components/discount/index.tsx

```diff
@@ -67,21 +67,22 @@ export const Discount = function Discount({ backgroundImage, discounts }: Props)
       />
       <button
         type="button"
-        onClick={() => setDismissed(true)}
+        onClick={() => {
+          setDismissed(true)
+        }}
         className="absolute right-5 top-5 text-foreground transition-transform hover:scale-110"
       >
         <X className="h-6 w-6" />
       </button>
 
-      <div
+      <button
         onClick={() => {
           if (isSpun) {
-            copy()
+            void copy()
```

Comment:
Is this `void` keyword necessary here?

Reply:
The `copy()` function is an async function that returns a Promise. ESLint's `@typescript-eslint/no-floating-promises` was warning about the unhandled promise since it can lead to unhandled rejections. Using `void` here indicates that we're aware this function returns a Promise, but we're intentionally not handling it - silencing the linter warning without changing the behavior of the code.

Let me know if you think there is a better way to handle this though! I'm open.

Reply:
I would usually opt for handling the promise with a `.catch` or `.then` but I think this is a reasonable approach 👍🏻